lib/ansible/module_utils/basic.py

```diff
@@ -1839,9 +1841,19 @@ def safe_eval(self, value, locals=None, include_exceptions=False):
     def _check_type_str(self, value):
         if isinstance(value, string_types):
             return value
-        # Note: This could throw a unicode error if value's __str__() method
-        # returns non-ascii.  Have to port utils.to_bytes() if that happens
-        return str(value)
+
+        # Ignore, warn, or error when converting to a string.
+        # The current default is to warn. Change this in Anisble 2.12 to error.
+        common_msg = 'The value {0!r} (type {0.__class__.__name__}) in a string field was converted to {1!r} (type string). {2}'
+        if self._string_conversion_action == 'error':
+            extra_msg = 'Quote the entire value to ensure it does not change.'
+            msg = common_msg.format(value, to_text(value), extra_msg)
+            raise TypeError(msg)
```

Comment:
The message is now maybe a bit too verbose, since it will be prepended by `argument %s is of type %s and we were unable to convert to %s: ` (i.e. it contains the type twice, and the information that it should be converted to a string).

Reply:
I was thinking that too. I'll refine it a bit.